viam-cartographer/src/carto_facade/util.cc

```diff
@@ -92,17 +104,19 @@ int read_pcd(std::string pcd, pcl::PCLPointCloud2 &blob) {
                            << data_type;
                 res = -1;
         }
+        if (res != 0) {
+            LOG(ERROR) << "Failed to parse PCD body";
+            return res;
```

Comment:
Would it be better to return directly in the places above where we set res?

Reply:
done